pkg/api/v1alpha1/machinedeploymentupgrade_types.go

```diff
@@ -18,6 +18,9 @@ type MachineDeploymentUpgradeSpec struct {
 
 	// KubernetesVersion refers to the Kubernetes version to upgrade the control planes to.
 	KubernetesVersion string `json:"kubernetesVersion"`
+
+	// MachineSpec is a base64 encoded json string value of the machineDeplopyment.Spec.Template.Spec field that's specification of the desired behavior of the machine.
+	MachineSpec string `json:"machineSpec"`
```

Comment:
Let's call this `MachineSpecData` to be inline with the changes we decided for cpupgrade